src/Vtex/Provider.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace SocialiteProviders\Vtex;
+
+use Laravel\Socialite\Two\AbstractProvider;
+use Laravel\Socialite\Two\ProviderInterface;
+use Laravel\Socialite\Two\User;
+
+class Provider extends AbstractProvider implements ProviderInterface
+{
+    /**
+     * Get the authentication URL for the provider.
+     *
+     * @param string $state
+     *
+     * @return string
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase(
+            $this->appendStorefrontDomain('/_v/oauth2/auth'),
+            $state
+        );
+    }
+
+    /**
+     * Get the POST fields for the token request.
+     *
+     * @param string $code
+     *
+     * @return array
+     */
+    protected function getTokenFields($code)
+    {
+        return array_merge(
+            parent::getTokenFields($code),
+            [
+                'response_type' => 'code',
+            ]
+        );
+    }
+
+    /**
+     * Get the token URL for the provider.
+     *
+     * @return string
+     */
+    protected function getTokenUrl()
+    {
+        return $this->appendStorefrontDomain('/_v/oauth2/token');
+    }
+
+    /**
+     * Get the raw user for the given access token.
+     *
+     * @param string $token
+     *
+     * @return array
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get(
+            $this->appendStorefrontDomain('/_v/oauth2/introspect'),
+            ['body' => 'token='.$token]
```

Comment:
Can you check https://developers.vtex.com/vtex-rest-api/docs/login-integration-guide-webstore-oauth2 ?